pkg/networkserver/mac_link_check.go

```diff
@@ -46,7 +47,12 @@ func handleLinkCheckReq(ctx context.Context, dev *ttnpb.EndDevice, msg *ttnpb.Up
 	gtws := make(map[string]struct{}, len(msg.RxMetadata))
 	maxSNR := msg.RxMetadata[0].SNR
 	for _, md := range msg.RxMetadata {
-		gtws[unique.ID(ctx, md.GatewayIdentifiers)] = struct{}{}
+		switch s := md.GetSource().(type) {
+		case *ttnpb.RxMetadata_GatewayIDs:
+			gtws[unique.ID(ctx, s.GatewayIDs)] = struct{}{}
+		case *ttnpb.RxMetadata_PacketBroker:
+			gtws[fmt.Sprintf("%s@%s/%s", s.PacketBroker.ForwarderID, s.PacketBroker.ForwarderNetID, s.PacketBroker.ForwarderTenantID)] = struct{}{}
```

Comment:
Is this the only occurence when this is helpful? Perhaps we want to extract this into a separate function?

Reply:
It is the only occurrence I think.

There was something similar in observability, but that wasn't consistent with the logging of duplication count, so I made that consistent by taking the `RxMetadata` length. I think that looking at that length makes sense for deduplication, and looking really at the unique gateways for coverage is here only.

Reply:
FTR; I extracted this logic because we also use it in metrics.